src/__tests__/components/fields/location-field/location-field.spec.tsx

```diff
@@ -1754,4 +1758,66 @@ describe("location-input-group", () => {
 			});
 		});
 	});
+
+	describe("Permission Modal events", () => {
+		it("should hide location modal for strict location", async () => {
+			renderComponent({
+				withEvents: true,
+				locationDetails: {
+					errors: {
+						generic: "throw something",
+					},
+				},
+				eventType: ELocationInputEvents.CLICK_OK_PERMISSION,
+				eventListener: (formRef: IFrontendEngineRef) => (e) => {
+					formRef.dispatchFieldEvent(ELocationInputEvents.CLOSE_LOCATION_MODAL, COMPONENT_ID);
+				},
+			});
+
+			await waitFor(() => window.dispatchEvent(new Event("online")));
+
+			getLocationInput().focus();
+
+			await waitFor(() => {
+				expect(getCurrentLocationErrorModal(true)).toBeInTheDocument();
+			});
+
+			within(getCurrentLocationErrorModal(true)).getByRole("button").click();
+
+			await waitFor(() => {
+				expect(getCurrentLocationErrorModal(true)).not.toBeInTheDocument();
+				expect(getLocationModal(true)).not.toBeInTheDocument();
+			});
+		});
```

Comment:
is it just me or is this formatting a bit off?

Reply:
looks ok in this comment box. but please double check

Reply:
<img width="708" alt="image" src="https://github.com/LifeSG/web-frontend-engine/assets/122194518/3ecc4fcb-68a9-454a-8e3b-6a2f567fe818">

Looks ok on vscode